### ISSUE TITLE
stat module sucks; getpwuid/getgrgid can raise KeyError

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5105]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#404|
|Type|defect|
|Created|2003-11-19 06:55:47Z|

Attachments:

* [ftp_callWithDefault-patch.txt](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/c35/c35a9fc52bb556c79f8fa540df587a2bf465b940/6a8cda6bc475c3eafb1dbe2eede5e20e77690a8e.txt) (2433 bytes) - added by exarkun on 2003-11-19 06:55:46Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__404 404
type__defect defect
reporter__exarkun exarkun
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1069224947000000 1069224947000000
changetime__1469403510645867 1469403510645867
version__ 
owner__slyphon slyphon
cc__exarkun cc__slyphon
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
Attached patch converts getUnixLongListString to not use the stat module, and to
catch KeyErrors possibly raised by getpwuid/getgrgid.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|GitHub commented|
|-|-|

In changeset 1e5d96a7c9ee8ce7fd4e798b4a877f169ad0e686
```
#!CommitTicketReference repository="" revision="1e5d96a7c9ee8ce7fd4e798b4a877f169ad0e686"
Merge pull request #404 from alex/patch-1

Author: alex
Reviewer: hawkowl
Fixes: #8276
```
